tools/test

```diff
@@ -1,4 +1,21 @@
 #!/usr/bin/env bash
+
+# Careful! This doesn't do everything you'd think it does. In fact, we
+# don't get its benefit in any of the `run_foo` functions. That's
+# because our logic to catch a failing suite, add it to a list, and
+# continue to the other suites, puts a bunch of commands in a list
+# before a `||`. That puts them in a special case, as the Bash manual
+# says:
+#
+# """
+# The shell does not exit if the command that fails is [...] part of
+# any command executed in a `&&` or `||` list except the command
+# following the final `&&` or `||` [...].
+# """
```

Comment:
nit: I would use:

```
> blockquote style
```

rather than the triple quotes, since I think that's clearer, and uses less vertical space.